include/dr/mhp/views/views.hpp

```diff
@@ -16,11 +16,18 @@ template <typename R> auto local_segments(R &&dr) {
   auto is_local = [](const auto &segment) {
     return dr::ranges::rank(segment) == default_comm().rank();
   };
+
   // Convert from remote iter to local iter
   auto local_iter = [](const auto &segment) {
-    auto b = dr::ranges::local(rng::begin(segment));
-    return rng::subrange(b, b + rng::distance(segment));
+    if constexpr (localizable_range<decltype(*rng::begin(segment))>) {
```

Comment:
Can you explain what is happening here?

Reply:
This _if_ checks if segment consists of ranges which can be made local. This is the case of sliding_view which consists of not elements (that can be made local) but ranges of elements (that can be made local).

Previously there was _is_sliding_view_ here, but IMO the current form is better as it is more general.